### PR TITLE
tests: change ostree boot location for RHEL 8.4

### PR DESCRIPTION
**Pull request by jrusz**
Change BOOT_LOCATION to COMPOSE_URL until GA is released.

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Comment by ondrejbudai** [comment]
I think that we should just remove 8.4 from the test suite, I wanted to wait until #1351 is merged but I think we might want to merge the removal earlier.

**Comment by jrusz** [comment]
> I think that we should just remove 8.4 from the test suite, I wanted to wait until #1351 is merged but I think we might want to merge the removal earlier.

We still want to at least use it for testing 8.4 RC 1.2 compose qualification...

**Comment by ondrejbudai** [comment]
Then the change should be done in the `rhel-8.4.0` branch I think.

File: test/cases/ostree.sh
Comment (atodorov) [inline test/cases/ostree.sh:36]: `BOOT_LOCATION="$COPOSE_URL/compose/BaseOS/x86_64/os/"`

File: test/cases/ostree.sh
Comment (atodorov) [inline test/cases/ostree.sh:33]: Can you also change the value of line 31 above to download.devel and remove the comment above it.

**Review by atodorov — CHANGES_REQUESTED** [review]

**Comment by atodorov** [comment]
> I think that we should just remove 8.4 from the test suite, I wanted to wait until #1351 is merged but I think we might want to merge the removal earlier.

Why do you want to remove 8.4 ? Is is because you want to replace with 8.5 or because of other reasons ?

**Comment by atodorov** [comment]
> Then the change should be done in the `rhel-8.4.0` branch I think.

We aren't (probably) going to release this change immediately but don't we need an approved RHBZ to merge into the branch ?

**Comment by atodorov** [comment]
@henrywang FYI.

File: test/cases/ostree.sh
Comment (ondrejbudai) [inline test/cases/ostree.sh:34]: I think I'm confused by this PR - `COMPOSE_URL` isn't defined in `ostree.sh`, how is this going to work on a PR pipeline?

Comment (atodorov) [reply test/cases/ostree.sh:34]: Indeed you are right. This is at the top of prepare-rhel-internal.sh -> "${COMPOSE_URL:-http://download.devel.redhat.com/rhel-8/nightly/RHEL-8/$COMPOSE_ID}"

I think we need to move this either as a separate script or define it in Jenkinsfile.  Or default to nightly here if COMPOSE_URL isn't set.

Comment (jrusz) [reply test/cases/ostree.sh:34]: Jenkinsfile would work, but don't know what value to put there as default... I guess separate script would work better as it could be called whenever and set the variable?

**Comment by henrywang** [comment]
Thanks @atodorov. One more question, is `COMPOSE_URL` defined by CI or ?

**Comment by atodorov** [comment]
> Thanks @atodorov. One more question, is `COMPOSE_URL` defined by CI or ?

```
$ grep COMPOSE_URL * -R
schutzbot/prepare-rhel-internal.sh:COMPOSE_URL="${COMPOSE_URL:-http://download.devel.redhat.com/rhel-8/nightly/RHEL-8/$COMPOSE_ID}"
schutzbot/prepare-rhel-internal.sh:# in case COMPOSE_URL was defined from the outside refresh COMPOSE_ID file,
schutzbot/prepare-rhel-internal.sh:curl -L "$COMPOSE_URL/COMPOSE_ID" > COMPOSE_ID
schutzbot/prepare-rhel-internal.sh:baseurl=${COMPOSE_URL}/compose/BaseOS/${ARCH}/os
schutzbot/prepare-rhel-internal.sh:baseurl=${COMPOSE_URL}/compose/AppStream/${ARCH}/os
schutzbot/prepare-rhel-internal.sh:            "baseurl": "${COMPOSE_URL}/compose/BaseOS/${ARCH}/os",
schutzbot/prepare-rhel-internal.sh:            "baseurl": "${COMPOSE_URL}/compose/AppStream/${ARCH}/os",
schutzbot/prepare-rhel-internal.sh:            "baseurl": "${COMPOSE_URL}/compose/RT/${ARCH}/os",
schutzbot/prepare-rhel-internal.sh:    "${COMPOSE_URL}/compose/AppStream/${ARCH}/os/Packages/"
test/README.md:flavors (e.g. rel-eng) specify the `COMPOSE_URL` environment variable to point to
test/README.md:!['Define COMPOSE_URL'](./define_compose_url.png)
```

**Comment by atodorov** [comment]
This is blocked by #1393

**Review by atodorov — APPROVED** [review]

**Pull request merged**
